aws/cloudformation/cloud_formation_stack.yml.erb

```
@@ -515,6 +515,7 @@ Resources:
       InstanceType: !Ref ConsoleInstanceType
       IamInstanceProfile: !Ref FrontendInstanceProfile
       KeyName: <%=SSH_KEY_NAME%>
+      DisableApiTermination: <%= rack_env?(:adhoc) ? false : true %>
```

Any reason not to do
```suggestion
      DisableApiTermination: <%= !rack_env?(:adhoc) %>
```

---

Are there any string vs boolean issues here? I don't remember whether that's a concern with erb into yml or not.

---

Possibly! But if there are issues like that, then those issues will manifest in the ternary operator version, too.

---

I think `!rack_env?(:adhoc` is better syntax, but given difficulty testing some of these changes (provision a "full stack" adhoc is unreliable), I'm staying with `<%= rack_env?(:adhoc) ? false : true %>` because that's the syntax we've used in a couple of other places in this template.

Note that there's a flaw in our usage of `rack_env?` - when provisioning a new environment, `rack_env?` identifies the environment of the engineer's workstation (which we typically override with `RAILS_ENV=adhoc` when issuing the rake command. We should transition to specifying the `environment` for a system we're provisioning with this template via a CloudFormation template Parameter instead.